mesa/mesaPROTON_OC.py

```diff
@@ -294,21 +308,31 @@ def soc_add_more_friends(self, targets):
                 target.makeFriends(extra.weighted_one_of(support_set, lambda x: max_criminal_tendency - x.criminal_tendency, self.rng))
 
     def welfare_intervene(self):
-        if welfare_support == "job_mother":
-            targets = [x for x in schedule.agents if
-                       x.gender_is_male == False and
-                       not x.my_job and
-                       (True if not x.neighbors.get("partner") else not x.neighbors.get("partner").pop().oc_member)
-                       ]
-        if welfare_support == "job_child":
-            targets = [x for x in schedule.agents if
-                       x.age() > 16 and x.age() < 24 and
-                       not x.my_school and
-                       not x.my_job and
-                       x.father.oc_member
-                       ]
+        """
+        This procedure is active when the self.welfare_support attribute is different from None. There are 2
+        possible welfare interventions: job-mother or job-child. These parameters determine the portion of the population
+        on which the intervention will have effect. The intervention consists of the application of a single procedures
+        on eligible family members:
+        1. new jobs are created and assigned to eligible members (mothers or children)
+        :return: None
+        """
+        if self.welfare_support == "job-mother":
+            targets = list()
+            for mother in [agent.mother for agent in self.schedule.agents if agent.mother]:
+                if not mother.my_job and mother.neighbors.get("partner"):
+                    if mother.get_link_list("partner")[0].oc_member:
+                        targets.append(mother)
+
+        if self.welfare_support == "job-child":
+            targets = list()
+            for agent in self.schedule.agents:
+                if agent.age() > 16 and agent.age() < 24 and not agent.my_school and not agent.my_job and agent.father:
```

Comment:
Let's put >= 16 here too, for uniformity? They can work when they reach 16, right?

Reply:
Yes, I think it makes sense to create an attribute here: model.age_enter_labor_market. what do you think?

Reply:
I support the attribute.

Reply:
Great!